R/dataframe-group_by-general.R

```diff
@@ -35,11 +35,18 @@ wrap_to_group_by <- function(x, by, maintain_order) {
 #'   c_mean_squared = (pl$col("c") ** 2)$mean()
 #' )
 groupby__agg <- function(...) {
-  self$df$lazy()$group_by(
-    !!!self$by,
-    .maintain_order = self$maintain_order
-  )$agg(...)$collect(optimizations = QueryOptFlags()$no_optimizations()) |>
-    wrap()
+  wrap({
+    out <- self$df$lazy()$group_by(
+      !!!self$by,
+      .maintain_order = self$maintain_order
+    )
+
+    if (!is.null(self$predicates)) {
+      out <- out$having(!!!self$predicates)
+    }
+
+    out$agg(...)$collect(optimizations = QueryOptFlags()$no_optimizations())
+  })
```

Comment:
In the python polars PR, they create a helper function `_lgb()` but that doesn't seem very useful to me.